test/test_tensor_creation_ops.py

```diff
@@ -832,6 +832,139 @@ def compare_helper_(like_fn, t):
                     tp = t.permute(p)
                     compare_helper_(like_fn, tp)
 
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False))
+    def test_hsplit(self, device, dtype):
+        inputs = (
+            ((), 3),
+            ((), [2, 4, 6]),
+            ((6,), 2),
+            ((6,), 4),
+            ((6,), [2, 5]),
+            ((6,), [7, 9]),
+            ((3, 8), 4),
+            ((3, 8), 5),
+            ((3, 8), [1, 5]),
+            ((3, 8), [3, 8]),
+            ((5, 5, 5), 2),
+            ((5, 5, 5), [1, 4]),
+            ((5, 0, 5), 3),
+            ((5, 5, 0), [2, 6]),
+        )
+
+        for shape, arg in inputs:
+            t = make_tensor(shape, device, dtype)
+            t_n = t.cpu().numpy()
+
+            if len(shape) < 1:
+                with self.assertRaisesRegex(RuntimeError, "torch.hsplit requires a tensor with at least 1 dimension"):
+                    torch.hsplit(t, arg)
+                with self.assertRaises(ValueError):
+                    np.hsplit(t_n, arg)
+
+            elif isinstance(arg, list) or len(shape) == 0:
+                result = torch.hsplit(t, arg)
+                result_n = np.hsplit(t_n, arg)
+                self.assertEqual(result, result_n)
+            else:
```

Comment:
What is this else block catching? Add a comment here explaining OR, better, make this another conditional `elif` and then have an `else` block that throws an error (because something unexpected passed through)